### PR TITLE
feat: polish `StrictSubstitution.apply`

### PR DESCRIPTION
**Pull request by JonathanStarup**
- apply the substitution to the result of resolving associated types (to ensure normalized types)
- apply the substitution to the result of s.m (to ensure normalized types)
- remove unnecessary nesting in helper function
- Actually remove aliases

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala [hunk 1]
```diff
@@ -79,46 +79,40 @@ object MonoDefs {
...
-            val arg = visit(arg0)
-            EqualityEnvironment.reduceAssocType(cst, arg, eqEnv) match {
-              case Ok(t) => t
-              case Err(_) => throw InternalCompilerException("unexpected associated type reduction failure", loc)
-            }
+    def apply(tpe0: Type): Type = tpe0 match {
+      case x: Type.Var =>
+        // Remove variables by substitution, otherwise by default type.
+        s.m.get(x.sym) match {
+          case Some(t) => t.map(default) // TODO the map of default should not be needed if s is normalized
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala:86]: Why todo?

File: main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala [hunk 2]
```diff
@@ -79,46 +79,40 @@ object MonoDefs {
...
+          case None => default(tpe0)
+        }
+      case Type.Cst(_, _) =>
+        tpe0
+      case Type.Apply(t1, t2, loc) =>
+        val y = apply(t2)
+        apply(t1) match {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala:93]: Why this form? We can apply to both t1 and t2 and pattern match.

**Comment by magnus-madsen** [comment]
It says polish, but I don't see many new comments that clarifies :) :)

File: main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala [hunk 3]
```diff
@@ -79,46 +79,40 @@ object MonoDefs {
       *
       * NB: Applies the substitution first, then replaces every type variable with the unit type.
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala:80]: I guess this documentation and the documentation for `StrictSubstitution` is no longer accurate.

**Comment by JonathanStarup** [comment]
Oh yeah sorry for the confusin, I have the other PR with documentation updates

File: main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala [hunk 4]
```diff
@@ -120,46 +121,44 @@ object MonoDefs {
...
-            }
+    def apply(tpe0: Type): Type = tpe0 match {
+      case x: Type.Var =>
+        // Remove variables by substitution, otherwise by default type.
+        s.m.get(x.sym) match {
+          case Some(t) =>
+            // The image of s might have unconstrained variables and be non-normalized.
+            // It is important to use default first, because the variables of t might be in the
+            // domain of the substitution.
+            apply(t.map(default))
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala:132]: First important change, apply must be used for normalization (doesn't do anything right now)

**Review by mlutze — APPROVED** [review]

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**